### DISCUSSION TITLE
invalid bottle URL for updated formula, homebrew-freecad tap

### DISCUSSION THREAD

**Discussion by ipatch** [Homebrew/discussions#2939] [Tap maintenance and Homebrew development] [answered]
i added support for macos mojave recently via our `tests.yml` file thanks to a prior discussion, and setting up an additional self-hosted runner.  all three of the self-hosted runners built successful bottles for the swig formula we maintain. note that this formula has a `@` in the formula name if that makes a difference, and i do have the formula labeled as [keg only and versioned](https://github.com/FreeCAD/homebrew-freecad/blob/204e71592c19b488965287484b6723191b711bfa/Formula/swig%404.0.2.rb#L28).

i added a [revision block/statement](https://github.com/FreeCAD/homebrew-freecad/blob/204e71592c19b488965287484b6723191b711bfa/Formula/swig%404.0.2.rb#L8) to the formula file to allow the CI to rebuild swig, and all runners successfully rebuilt swig, and then i labeled the PR with the `pr-pull` label to package up and update the bottles / references for the bottles. and all of that seemed to go well and new bottles were published using the pr publish workflow in the tap.

so just testing things from a user perspective and not a maintainer of the homebrew-freecad tap, i run,

```
brew update
```

followed by

```
brew outdated
```

and see the below output,

```
freecad/freecad/swig@4.0.2 (4.0.2) < 4.0.2_1
```

i then attempt to install the updated bottle that was just published to the hombrew-freecad tap

```
brew upgrade freecad/freecad/swig@4.0.2
```

and get the below output

```
==> Downloading https://github.com/FreeCAD/homebrew-freecad/releases/download/swig@4.0.2-4.0.2/swig%404.0.2-4.0.2_1.catalina.bottle.2.tar.gz
##O#- #
curl: (22) The requested URL returned error: 404
Error: swig@4.0.2: Failed to download resource "swig@4.0.2"
Download failed: https://github.com/FreeCAD/homebrew-freecad/releases/download/swig@4.0.2-4.0.2/swig%404.0.2-4.0.2_1.catalina.bottle.2.tar.gz
```

is this something on the homebrew-freecad tap maintainer's end 🙋‍♂️ or is there something upstream that slipped in that shouldn't be there in _pr pull_ or with brew _test-bot_?

if it's something i ended foobarring, how would i go about preventing this from happening again? and possibly what would be the fix?

---

> now that i've typed all of this out, i did add a revision block and updated some other attributes ie. license block of the formula in the same commit if that makes a difference. 🦆

**Comment by carlocab** [comment, marked as answer]
Seems like a transient network issue to me:
```
linuxbrew@6bfa34b9febf:~$ brew fetch --bottle-tag=catalina freecad/freecad/swig@4.0.2
==> Fetching swig@4.0.2 from freecad/freecad
==> Downloading https://github.com/FreeCAD/homebrew-freecad/releases/download/swig@4.0.2-4.0.2_1/swig%404.0.2-4.0.2_1.catalina.bottle.tar.gz
==> Downloading from https://objects.githubusercontent.com/github-production-release-asset-2e65be/18778796/25265de5-1ce5-401a-91f9-999c2e9f5ac2?X-Amz-
######################################################################## 100.0%
Downloaded to: /home/linuxbrew/.cache/Homebrew/downloads/3bffb4c77a1f7a63ef79f7f283d05283c3b0fc16128ade9fbbcc78962e8a94a4--swig
SHA256: 958260091ff65decf4b291c959cd10b7f0d5572cd2e517ac0a70ab931cf23bd5
```
That is, unless, you've done something to fix this in the meantime.

**Reply by ipatch** [reply]
i was working a feature branch, and never switched my repo back to master 🤦‍♀️, and seems curl is now able to reach the endpoint. 👍

**Reply by carlocab** [reply]
`brew doctor` should've warned you about that. `brew doctor` is usually worth a try when you see any `brew`-related issues.

**Comment by carlocab** [comment]
Oh, wait. You were trying to download `https://github.com/FreeCAD/homebrew-freecad/releases/download/swig@4.0.2-4.0.2/swig%404.0.2-4.0.2_1.catalina.bottle.2.tar.gz` while I successfully downloaded `https://github.com/FreeCAD/homebrew-freecad/releases/download/swig@4.0.2-4.0.2_1/swig%404.0.2-4.0.2_1.catalina.bottle.tar.gz`.

Did you manually add a `rebuild` to your bottle block? If so, don't do that. `test-bot` manages `rebuild`s. (Or any other part of the `bottle` block.)

**Reply by ipatch** [reply]
i don't i ever manually added a rebuild block to swig.

of course, i've slept several times 😴 since working with this CI/repo
